### ISSUE TITLE
[FR] Export internal links within file

### ISSUE THREAD

**Issue by RyotaUshio** [l1xnan/obsidian-better-export-pdf#100]
First of all, thank you very much for this awesome plugin!

It would be great if this plugin could support exporting internal links within the same file to PDFs.

For example, suppose we want to export `note.md`, which looks like this, to PDF.
```
[[#heading]]
[[note#heading]]

# heading
content
```
The two links target the heading in the same file, so it would be very convenient if we could click those links to jump to the heading in the exported PDF. The same applies to block links.

**Comment by l1xnan** [comment]
Thanks to your FR, this is a necessary feature point that is supported in the latest version(1.5.0).

**Comment by RyotaUshio** [comment]
Hi, thank you very much for making this happen!
It works very well for heading links, which is great.

It would be even better if block links could be supported as well although I'm not sure how difficult it would be.

**Comment by RyotaUshio** [comment]
I see. In fact, I've encountered a similar problem when I was writing my plugin, and found a way to associate a rendered div and the corresponding `SectionCache`.

I'll leave my code here for reference, but it's not good as an example so I will explain in more detail later when I have time.

https://github.com/RyotaUshio/obsidian-latex-theorem-equation-referencer/blob/978f8dc28459fb8c96bf5b732c0f0fa301e2eeea/src/equations/reading-view.ts#L37

**Comment by l1xnan** [comment]
I released version `1.5.1` with a rudimentary implementation of the block links feature.

**Comment by RyotaUshio** [comment]
Thank you! Unfortunately, 1.5.1 seems to have some issues when a callout is followed by its block ID.

For example,

```
> [!NOTE] Title

^block1

> [!NOTE] Title
> Content

^block2
```

results in

<img width="479" alt="image" src="https://github.com/l1xnan/obsidian-better-export-pdf/assets/72342591/1f04b25f-1acc-4199-af17-8eacac5dddfc">

- The first callout is not rendered
- The second line of the other callout is rendered with `>` prepended

**Comment by l1xnan** [comment]
It's true that `span.block-id` is causing the problem, and I've added a `\n` symbol to fix it and released version `1.5.2`.
